Tests/src/PublicAPITests/UserSystemTests.cpp

```diff
@@ -1010,3 +1010,19 @@ CSP_PUBLIC_TEST(CSPEngine, UserSystemTests, GetGuestProfileTest)
 	LogOut(UserSystem);
 }
 #endif
+
+#if RUN_ALL_UNIT_TESTS || RUN_USERSYSTEM_TESTS || RUN_USERSYSTEM_AGE_NOT_VERIFIED_TEST
+CSP_PUBLIC_TEST(CSPEngine, UserSystemTests, AgeNotVerifiedTest)
+{
+	auto& SystemsManager = csp::systems::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+
+	csp::common::String UserId;
+
+	// Log in
+	auto [Result]
+		= Awaitable(&csp::systems::UserSystem::Login, UserSystem, "", DefaultLoginEmail, DefaultLoginPassword, false).Await(RequestPredicate);
+
+	EXPECT_EQ(Result.GetResultCode(), csp::services::EResultCode::Failed);
```

Comment:
This test should replicate the JS test by testing both `false` and `true` values.

Reply:
I believe we also have a C++ helper function for logging in, which should be used instead.

Reply:
sorted